### PR TITLE
WIP: Use async/await in ENS hook

### PR DESCRIPTION
**Pull request by pcowgill**

### CODE REVIEW

**Comment by pcowgill** [comment]
@NoahZinsmeister Sorry, this change was more subtle than I realized - still getting used to hooks since I mostly do React Native when working on front-end, where hooks arrived later.

This particular PR wasn't to fix an issue, but rather in an effort to use async/await syntax consistently throughout the codebase (with the idea being that consistent style helps to prevent future issues, by making sure new contributors follow the right error-handling patterns, etc.).

I like general direction of your [latest commit](https://github.com/Uniswap/uniswap-frontend/pull/486/commits/a9736c613ca9ac012c5ec4acac7cf46eacf27850), but I think (A) we can make it clearer what's being returned to cleanup the effect by splitting that out and (B) we can still achieve my goal of no vanilla promise syntax (which admittedly you don't have as a stated goal necessarily, so you'd have to buy my rationale above - this probably should have started in a GitHub issue, which I'll very much note going forward).

There is a way to use async/await syntax in `useEffect` while still cleaning up using the `return` feature of the hook.

I'm thinking this style will work well:

```
const [data, setData] = useState();
useEffect(() => {
  let cancel = false;
  const runEffect = async () => {
    const data = await fetchSomeData(filter);
    if (cancel) {
      return;
    }
    setData(data);
  };
  runEffect();

  // Cleanup function that will be called on
  // 1. Unmount
  // 2. Dependency Array Change
  return () => {
    cancel = true;
  }
}, [setData, filter]);
```

Credit to [this blog post](https://dev.to/n1ru4l/homebrew-react-hooks-useasynceffect-or-how-to-handle-async-operations-with-useeffect-1fa8), although I think the eventual conclusion in that post of using generators or a special hook is overkill here since there's just one async function being used.

**Comment by pcowgill** [comment]
> I'll note that this ^ style doesn't cancel the fetch itself though (that blog post talks about that in more detail) - not clear to me yet whether canceling the fetch is happening in the current `beta` branch of this repo or not, in which case the proposed change above would be a change for the worse from a performance perspective.

@NoahZinsmeister I don't think the current approach with promises cancels the fetch once it has been initiated either. So I'd suggest going with the style I proposed above. If you're happy with how that looks, let me know and I'll make another commit to this branch that modifies this hook to follow that style. Thanks!

**Comment by pcowgill** [comment]
If it's simpler to reason about by looking at the exact changes that would be made, I'd be happy to make another commit to this feature branch.

**Comment by NoahZinsmeister** [comment]
@pcowgill re. canceling, you're right that it's not possible, since we're using native promises we have to settle for ignoring the output. your code looks correct, and i agree that consistency is good! however, if applied across the codebase this would be a fairly substantial change for a pattern that, as far as i can tell, isn't particularly broken.

i guess i'm struggling to understand the motivation behind wanting to use async syntax in all `useEffect`s, as opposed to just using promise syntax. imo the boilerplate for supporting async is actually more verbose than just using `.catch`. in this particular case i definitely agree there's some wonkiness going on (the [catch here](https://github.com/Uniswap/uniswap-frontend/blob/beta/src/hooks/index.js#L61) for example, is i believe catching errors in the (sync) portion of `lookupAddress`? i remember this being a very atypical case).

long story short, i'm ok changing this particular instance to something like the code you cited _if_ it fixes a demonstrated bug (which i still am unable to replicate), but i'm not convinced that applying this across the codebase makes sense.

**Comment by pcowgill** [comment]
@NoahZinsmeister The motivation is just for consistency - I think newer js devs don't necessarily even fully understand promises (putting aside arguments that they should for now), but they do understand async/await. That means that using the promises API leaves the potential open for future bugs, unless the person reviewing PRs is watchful.

You're right that the boilerplate for async/await in this case is more verbose.

That [example you pointed out](https://github.com/Uniswap/uniswap-frontend/blob/beta/src/hooks/index.js#L61) is a good one - you get used to using try/catch in the rest of the codebase with async/await, but then mixing try/catch and promises gets very hard to reason about because that try/catch won't catch what the `.catch` for the promise would have caught. So it can be easy to forget the `.catch` that's needed for proper error-handling in a case like that.

This isn't a bug fix PR, just a style-that-makes-it-harder-to-add-potential-future-bugs PR. (Separately, it is worth adding a `.catch` for the `library.lookupAddress`, though.) 

So given all of that, keeping it as-is is a very reasonable choice! I'll close this PR.
